code/modules/projectiles/guns/lever_action.dm

```diff
@@ -327,6 +347,142 @@ their unique feature is that a direct hit will buff your damage and firerate
+	lever_delay = FIRE_DELAY_TIER_3
+	accuracy_mult = BASE_ACCURACY_MULT + HIT_ACCURACY_MULT_TIER_5
+	accuracy_mult_unwielded = BASE_ACCURACY_MULT - HIT_ACCURACY_MULT_TIER_10
+	scatter = SCATTER_AMOUNT_TIER_8
+	burst_scatter_mult = 0
+	scatter_unwielded = SCATTER_AMOUNT_TIER_2
+	damage_mult = BASE_BULLET_DAMAGE_MULT
+	recoil = RECOIL_AMOUNT_TIER_3
+	recoil_unwielded = RECOIL_AMOUNT_TIER_1
+
+/obj/item/weapon/gun/lever_action/xm88/set_gun_attachment_offsets()
+	attachable_offset = list("muzzle_x" = 33, "muzzle_y" = 19, "rail_x" = 11, "rail_y" = 21, "under_x" = 24, "under_y" = 16, "stock_x" = 12, "stock_y" = 16)
+
+/obj/item/weapon/gun/lever_action/xm88/apply_hit_buff()
+	lever_sound = lever_super_sound
+	lever_message = "<b><i>You quickly press the [lever_name]!<i><b>"
+	last_fired = world.time - buff_fire_reduc //to shoot the next round faster
+	damage_mult = initial(damage_mult) + BULLET_DAMAGE_MULT_TIER_4
+
+	if(floating_penetration < floating_penetration_upper_limit)
+		floating_penetration++
+
+	for(var/slot in attachments)
+		var/obj/item/attachable/AM = attachments["slot"]
+		if(AM && (AM.damage_mod || AM.delay_mod))
+			damage_mult += AM.damage_mod
+			fire_delay += AM.delay_mod
+	wield_delay = 0 //for one-handed levering
+
+/obj/item/weapon/gun/lever_action/xm88/Fire(atom/target, mob/living/user, params, reflex, dual_wield)
+	if(!able_to_fire(user) || !target) //checks here since we don't want to fuck up applying the increase
+		return
+	if(floating_penetration && in_chamber) //has to go before actual firing
+		var/obj/item/projectile/P = in_chamber
+		switch(floating_penetration)
+			if(FLOATING_PENETRATION_TIER_1)
+				P.ammo = GLOB.ammo_list[/datum/ammo/bullet/lever_action/xm88/pen10]
+			if(FLOATING_PENETRATION_TIER_2)
+				P.ammo = GLOB.ammo_list[/datum/ammo/bullet/lever_action/xm88/pen20]
+			if(FLOATING_PENETRATION_TIER_3)
+				P.ammo = GLOB.ammo_list[/datum/ammo/bullet/lever_action/xm88/pen30]
+			if(FLOATING_PENETRATION_TIER_4)
+				P.ammo = GLOB.ammo_list[/datum/ammo/bullet/lever_action/xm88/pen40]
+		. = ..()
+		return
+	else
+		return ..()
+
+/obj/item/weapon/gun/lever_action/xm88/unload(mob/user)
+	if(levered)
+		to_chat(user, SPAN_WARNING("You open \the [src]'s breech and take out a round."))
+		levered = FALSE
+	return empty_chamber(user)
+
+/obj/item/weapon/gun/lever_action/xm88/reset_hit_buff(var/one_hand_lever) //why does this need a user arg when it doesn't use user at all?
+	if(!(flags_gun_lever_action & USES_STREAKS))
+		return
+	SIGNAL_HANDLER
+	streak = 0
+	lever_sound = initial(lever_sound)
+	lever_message = initial(lever_message)
+	wield_delay = initial(wield_delay)
+	cur_onehand_chance = initial(cur_onehand_chance)
+	if(in_chamber)
+		var/obj/item/projectile/P = in_chamber
+		P.ammo = GLOB.ammo_list[/datum/ammo/bullet/lever_action/xm88]
+	floating_penetration = FLOATING_PENETRATION_TIER_0
+	//these are init configs and so cannot be initial()
+	lever_delay = FIRE_DELAY_TIER_3
+	fire_delay = FIRE_DELAY_TIER_1
```

Comment:
This is very different from the delay in `set_gun_config_values()`. Intended?

Reply:
er probably not